### ISSUE TITLE
Incorrect numeric enum backing datatype - varchar/text

### ISSUE THREAD

**Issue by pirj** [djezzzl/database_consistency#103]
The problem only becomes apparent when `prepared_statements:` option is set to `false`. This is typical and [recommended by Rails when connecting via PgBouncer](https://guides.rubyonrails.org/configuring.html#configuring-a-postgresql-database).

```
ActiveRecord::StatementInvalid: PG::UndefinedFunction: ERROR:  operator
   does not exist: text = integer
      LINE 1: SELECT "users".* FROM "users" WHERE "users"."os" = 0
```

```ruby
# app/models/user.rb
enum os: [:linux, :windows]
```

```ruby
# db/schema.rb
  t.text "os"
```

The data actually saved in the column is `'1'`, `'2'` etc.

With `prepared_statements: true` this error doesn't happen, and it works as expected (well, not really expected, I would expect it to blow up in both cases, but there's an implicit conversion somewhere along the way in statement preparation).

Should be easy to detect for both
```ruby
enum os: [:linux, :windows]
```
and
```ruby
enum os: { linux: 1, windows: 2}
```
definition options.

**Comment by djezzzl** [comment]
Hi @pirj,

Sorry for taking me so long to look at the issue. I've tried it myself and was able to reproduce it.

Am I right that you're suggesting to type cast values on an enum to strings?

**Comment by pirj** [comment]
Hi @djezzzl !

I'd suggest changing the column type to an integer/smallint.

Or [from Rails API doc](https://api.rubyonrails.org/classes/ActiveRecord/Enum.html), 
> it's also possible to use a string column to persist the enumerated value. Note that this will likely lead to slower database queries:

```ruby
class Conversation < ActiveRecord::Base
  enum :status, active: "active", archived: "archived"
```

I'd argue about slower DB queries for a string enum type, but in this context, I just propose not to pay attention to this statement for decision-making.

**Comment by djezzzl** [comment]
Finally released in 1.4.10. 

Sorry for taking so long to implement it! And thank you for your suggestions!
